tests/Unit/CrudPanel/CrudPanelAccessTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Backpack\CRUD\Tests\Unit\CrudPanel;
+
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class CrudPanelAccessTest extends BaseCrudPanelTest
+{
+    private $unknownPermission = 'unknownPermission';
+
+    private $defaultAccessList = [
+        'list',
+        'create',
+        'update',
+        'delete',
+    ];
+
+    private $fullAccessList = [
+        'list',
+        'create',
+        'update',
+        'delete',
+        'revisions',
+        'reorder',
+        'show',
+        'details_row',
+    ];
+
+    public function testDefaultAccess()
+    {
+        $this->assertEquals($this->defaultAccessList, $this->crudPanel->access);
+    }
+
+    public function testHasAccess()
+    {
+        foreach ($this->defaultAccessList as $permission) {
+            $this->assertTrue($this->crudPanel->hasAccess($permission));
```

Comment:
Technically this doesn't really prove that `hasAccess` is working - it just proves that it might be written like:

```
public function hasAccess($permission)
{
    return true;
}
```